### PR TITLE
fix MyPy errors

### PR DESCRIPTION
**Pull request by sinisaos**
Based on #58

### CODE REVIEW

File: piccolo_api/crud/endpoints.py [hunk 1]
```diff
@@ -666,19 +675,19 @@ async def get_single(self, request: Request, row_id: int) -> Response:
...
                 ]
-                columns = columns + readable_columns
+                columns = [*columns, *readable_columns]
 
             row = (
                 await self.table.select(
                     *columns, exclude_secrets=self.exclude_secrets
                 )
-                .where(self.table.id == row_id)
+                .where(self.table._meta.columns[0] == row_id)
```
Comment (aminalaee) [inline piccolo_api/crud/endpoints.py:690]: And here too `table._meta.primary_key`.

File: piccolo_api/crud/endpoints.py [hunk 2]
```diff
@@ -713,7 +722,9 @@ async def put_single(
         }
 
         try:
-            await cls.update(values).where(cls.id == row_id).run()
+            await cls.update(values).where(
+                cls._meta.columns[0] == row_id
```
Comment (aminalaee) [inline piccolo_api/crud/endpoints.py:726]: And here too `table._meta.primary_key`.

File: piccolo_api/token_auth/middleware.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ async def get_user(self, token: str) -> User:
 
         user = (
             await self.auth_table.select(self.auth_table.username)
-            .where(self.auth_table.id == user_id)
+            .where(self.auth_table._meta.columns[0] == user_id)
```
Comment (aminalaee) [inline piccolo_api/token_auth/middleware.py:69]: And here too `table._meta.primary_key`.

File: piccolo_api/shared/auth/user.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ def auth_table(self) -> t.Type[Table]:
 
     @property
     def user_id(self) -> int:
-        return t.cast(int, self.user.id)
+        return t.cast(int, self.user._meta.columns[0])
```
Comment (aminalaee) [inline piccolo_api/shared/auth/user.py:27]: And here too `table._meta.primary_key`.

Comment (sinisaos) [reply piccolo_api/shared/auth/user.py:27]: @aminalaee Thanks. I'll fix that.

Comment (aminalaee) [reply piccolo_api/shared/auth/user.py:27]: @sinisaos this is great BTW. Now all of them have mypy.

Comment (sinisaos) [reply piccolo_api/shared/auth/user.py]: @aminalaee Thank you for your suggestion because I completely forgot about `table._meta.primary_key`

**Comment by dantownsend** [comment]
Thanks for this.

I made a couple of minor changes.

It's now possible to configure MyPy inside `pyproject.toml`. I did this to ignore some type errors around imports, so there are fewer `type: ignore` comments now.

Also, I had to add `piccolo[postgres,sqlite]` to `test-requirements.txt` as Piccolo no longer installs the db libraries by default.

**Pull request merged**